src/core/modal/layouts/link-picker/modal-layout-link-picker.element.ts

```diff
@@ -0,0 +1,223 @@
+import { css, html, nothing } from 'lit';
+import { UUITextStyles } from '@umbraco-ui/uui-css/lib';
+import { customElement, query, state } from 'lit/decorators.js';
+import { UUIBooleanInputEvent, UUIInputElement } from '@umbraco-ui/uui';
+import { UmbModalLayoutElement } from '../modal-layout.element';
+import { UmbTreeElement } from '../../../../backoffice/shared/components/tree/tree.element';
+import { buildUdi } from '@umbraco-cms/utils';
+
+export interface UmbModalLinkPickerData {
+	icon?: string;
+	name?: string;
+	published?: boolean;
+	queryString?: string;
+	target?: string;
+	trashed?: boolean;
+	udi?: string;
+	url?: string;
+}
+
+export interface UmbModalLinkPickerConfig {
+	hideAnchor?: boolean;
+	ignoreUserStartNodes?: boolean;
+	overlaySize?: 'small' | 'medium' | 'large' | 'full';
+}
+@customElement('umb-modal-layout-link-picker')
+export class UmbModalLayoutLinkPickerElement extends UmbModalLayoutElement<
+	UmbModalLinkPickerData & UmbModalLinkPickerConfig
+> {
+	static styles = [
+		UUITextStyles,
+		css`
+			hr {
+				border: none;
+				border-bottom: 1px solid var(--uui-color-divider);
+				margin-bottom: var(--uui-size-space-3);
+			}
+
+			uui-input,
+			uui-toggle,
+			uui-label {
+				width: 100%;
+			}
+
+			uui-input,
+			uui-label {
+				margin-bottom: var(--uui-size-space-6);
+			}
+
+			.url-link {
+				display: flex;
+				gap: var(--uui-size-space-6);
+			}
+			.url-link span {
+				flex: 1 1 0px;
+			}
+
+			#select-media {
+				display: block;
+			}
+		`,
+	];
+
+	@state()
+	_selectedKey?: string;
+
+	@state()
+	_link: UmbModalLinkPickerData = {
+		icon: undefined,
+		name: undefined,
+		published: true,
+		queryString: undefined,
+		target: undefined,
+		trashed: false,
+		udi: undefined,
+		url: undefined,
+	};
+
+	@state()
+	_layout: UmbModalLinkPickerConfig = {
+		hideAnchor: false,
+		ignoreUserStartNodes: false,
+	};
+
+	@query('#link-input')
+	private _linkInput!: UUIInputElement;
+
+	@query('#anchor-input')
+	private _linkQueryInput?: UUIInputElement;
+
+	@query('#link-title-input')
+	private _linkTitleInput!: UUIInputElement;
+
+	connectedCallback() {
+		super.connectedCallback();
+		this._link.icon = this.data?.icon;
+		this._link.name = this.data?.name;
+		this._link.published = this.data?.published ?? true;
+		this._link.queryString = this.data?.queryString;
+		this._link.target = this.data?.target;
+		this._link.trashed = this.data?.trashed ?? false;
+		this._link.udi = this.data?.udi;
+		this._link.url = this.data?.url;
```

Comment:
I think we can make one call to `this._link` to set it initially instead of a new call per line.

Reply:
The issue was i wanted to keep config and data separate in this file, but the data are sent together with config into `this.data`.

I think I found a better better way though